benchmark.exs

```diff
@@ -56,11 +57,11 @@ Benchee.run(%{
 # round-trip       20.40 K       49.02 μs    ±20.06%       46.53 μs       86.84 μs
 # new-alt          15.64 K       63.95 μs    ±14.60%       61.53 μs      104.48 μs
 
-# with latin1 conversion
-
-# raw              75.87 K       13.18 μs   ±113.53%       12.36 μs       24.25 μs
-# new              51.93 K       19.26 μs    ±28.73%       17.37 μs       38.47 μs
-# new-copy         47.77 K       20.93 μs    ±29.84%       19.21 μs       39.15 μs
-# round-trip       20.32 K       49.21 μs    ±18.52%       47.62 μs       80.37 μs
-# new-alt          15.50 K       64.53 μs    ±12.32%       62.72 μs       97.16 μs
+# add Maps output for use with the new HPath sigil
 
+# raw             137.72 K
+# new              43.00 K - 3.20x slower +15.99 μs
+# new-copy         38.38 K - 3.59x slower +18.80 μs
+# maps             32.75 K - 4.21x slower +23.27 μs
```

Comment:
Do we make up for the slower creation with the searching?

Do you have a benchmark on searching?

Reply:
I haven't done one. You're welcome to make some more if you want? I think the ergonomic benefits make it worthwhile even for the extra speed hit. There's likely some number of lookups that enumerate lists the old way we'd have to do before we'd hit break even. Probably a lot. But we're doing a lot now when you consider every event handler we have...

Reply:
```elixir
list_message = HL7.Message.new(msg)
map_message = HL7.parse!(list_message)

path = ~p{PID-18}
deep_path = ~p{PID-11.4}

Benchee.run(%{
  "list" => fn -> HL7.Query.find_first(list_message, path) end,
  "map" => fn -> HL7.get(map_message, path) end
})
```

```
Name           ips        average  deviation         median         99th %
list        3.07 M      326.00 ns ±11292.07%         208 ns         375 ns
map         2.69 M      372.17 ns  ±7779.30%         292 ns         458 ns

Comparison: 
list        3.07 M
map         2.69 M - 1.14x slower +46.17 ns

Name                ips        average  deviation         median         99th %
deep list        2.60 M      383.99 ns  ±9353.22%         250 ns         458 ns
deep map         2.49 M      402.21 ns  ±6922.81%         292 ns         500 ns

Comparison: 
deep list        2.60 M
deep map         2.49 M - 1.05x slower +18.21 ns
```

The map appears to be slower

Reply:
To note, slightly slower except for perhaps things like PV1-42 or such. Biggest benefit is for seeing index values in the maps.